services/api/src/utils/schema.js

```diff
@@ -85,6 +85,20 @@ function createSchema(attributes = {}, options = {}) {
     return this.save();
   };
 
+  schema.post(/^init|save/, function () {
+    if (!this.$locals.auditOriginal) {
+      this.$locals.auditOriginal = this.toObject({
+        depopulate: true,
+      });
+    }
+  });
+
+  // keeps the directModifiedPaths around after document.save
+  schema.pre('save', function () {
+    const paths = this.directModifiedPaths() || [];
+    this.$locals.auditPathsModified = uniq([...(this.$locals.pathsModified || []), ...paths]);
```

Comment:
we probably don't need to store the modified paths themselves just the snapshot on save... something like:

```js
  schema.pre('save', function () {
    const paths = this.directModifiedPaths();
    this.$locals.diff = {
      before: pick(this.$locals.original, paths),
      after: pick(this.toObject({ depopulate: true }), paths),
    };
  });

```